### PR TITLE
Qr design dashboard test

### PR DESCRIPTION
**Pull request by hritikaggarwal31**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "mobstac-awesome-qr",
-    "version": "2.1.5-beta.2",
+    "version": "2.1.7",
```
Comment (sourobrata) [inline package.json:3]: @hritikaggarwal31 This should be 2.1.5-beta.2 for the dashboard

File: package.json [hunk 2]
```diff
@@ -65,6 +65,8 @@
         "@types/node": "11.13.0",
         "canvas": "2.5.0",
         "svgdom": "^0.1.8",
-        "webpack-node-externals": "^2.5.2"
+        "webpack-node-externals": "^2.5.2",
+        "node-fetch": "^2.6.1",
```
Comment (sourobrata) [inline package.json:69]: @hritikaggarwal31 Are you using these dependencies? I think these break in the dashboard. Please remove them if not required.

File: src/Util.ts [hunk 3]
```diff
@@ -85,6 +85,25 @@ export const loadImage = (src: string, imageServerURL?: string | undefined, imag
     });
 };
 
+export const isSvgFile = (src: string) => {
```
Comment (sourobrata) [inline src/Util.ts:88]: @hritikaggarwal31 Is this required for your feature?

File: webpack.config.js [hunk 4]
```diff
@@ -15,6 +15,8 @@ module.exports = {
     resolve: {
         extensions: [ '.tsx', '.ts', '.js' ]
     },
+    externals: [{ fs: "commonjs fs" },
```
Comment (sourobrata) [inline webpack.config.js:18]: @hritikaggarwal31 Please remove this if not required.

File: src/Util.ts (hunk 3 above)
Comment (hritikaggarwal31) [reply src/Util.ts:88]: yes.

Comment (sourobrata) [reply src/Util.ts:88]: svg is not needed in the dashboard right? We always use png

**Review by sourobrata — APPROVED** [review]

File: src/Models.ts [hunk 5]
```diff
@@ -600,7 +605,106 @@ export class Drawing {
...
+            inX = true;
+        }
+        if(y> bottomY && y<topY) {
+            inY = true;
+        }
+       
+        return !(inX && inY);
+    }
+    private middleSquare(seed: number) {
+        const result = (seed * seed).toString().padStart(4,"0").slice(1, 3);
```
Comment (sourobrata) [inline src/Models.ts:625]: There is a chance this will fail in IE. Can we test it to confirm?

File: src/Svg.ts [hunk 6]
```diff
@@ -201,7 +201,22 @@ export class SVGDrawing {
                 return mainCanvas.svg();
             });
     }
-
+    private inShape(x: number, y: number,pt: number,side: number): boolean {
```
Comment (sourobrata) [inline src/Svg.ts:204]: Do we need this function in the dashboard branch?

Comment (hritikaggarwal31) [reply src/Svg.ts:204]: Yes, it is required.

**Review by sourobrata — APPROVED** [review]

**Review by sourobrata — APPROVED** [review]

**Pull request merged**